dvc/repo/artifacts.py

```diff
@@ -0,0 +1,52 @@
+import os
+from typing import TYPE_CHECKING, Any, Dict
+
+from dvc.annotations import Artifact
+from dvc.dvcfile import FileMixin
+from dvc.exceptions import DuplicatedArtifactError
+
+if TYPE_CHECKING:
+    from dvc.repo import Repo
+
+
+class ArtifactsFile(FileMixin):
+    from dvc.schema import ARTIFACTS_SCHEMA as SCHEMA
+
+    def dump(self, stage, **kwargs):
+        raise NotImplementedError
+
+    def merge(self, ancestor, other, allowed=None):
+        raise NotImplementedError
+
+
+class Artifacts:
+    repo: "Repo"
+
+    def __init__(self, repo) -> None:
+        self.repo = repo
+
+    def _read(self):
+        # merge artifacts from all dvc.yaml files found
+        artifacts: Dict[str, Dict[str, Any]] = {}
+        for (
+            dvcfile,
+            dvcfile_artifacts,
+        ) in self.repo.index._artifacts.items():  # pylint: disable=protected-access
+            # read the artifacts.yaml file if needed
+            if isinstance(dvcfile_artifacts, str):
+                dvcfile_artifacts = ArtifactsFile(
+                    self.repo,
+                    os.path.join(os.path.dirname(dvcfile), dvcfile_artifacts),
+                    verify=False,
+                ).load()
+            for name, value in dvcfile_artifacts.items():
+                if name in artifacts:
+                    # Q: maybe better to issue a warning here and take the first one?
```

Comment:
I think we should forbid conflictive names within the same `dvc.yaml` but we should not fail when names come from different `dvc.yaml` files.

In other top-level sections like `plots`, we allow having the same name on different `dvc.yaml` and differentiate by adding the path to dvc.yaml as prefix

Reply:
Good point, thanks @daavoo 
can you please read [the discussion here](https://github.com/iterative/gto/issues/198#issuecomment-1480901662) (hope that will be quick enough) and contribute your opinion?